src/qibosoq/programs/flux.py

```diff
@@ -71,6 +71,53 @@ def set_bias(self, mode: str = "sweetspot"):
             self.pulse(ch=flux_ch)
         self.sync_all(50)  # wait all pulses are fired + 50 clks
 
+    def find_qubit_sweetspot(self, pulse: Pulse):
+        """Return bias of a qubit from flux pulse."""
+        bias = 0.0
+        for qubit in self.qubits:
+            if pulse.dac == qubit.dac:
+                bias = qubit.bias if qubit.bias else 0
+        return bias
+
+    def execute_flux_pulse(self, pulse: Pulse):
+        """Fire a fast flux pulse the starts and ends in sweetspot."""
+        gen_ch = pulse.dac
+        max_gain = int(self.soccfg["gens"][gen_ch]["maxv"])
+        bias = self.find_qubit_sweetspot(pulse)
+        sweetspot = int(bias * max_gain)
+
+        duration = self.soc.us2cycles(pulse.duration, gen_ch=gen_ch)
+        samples_per_clk = self._gen_mgrs[gen_ch].samps_per_clk
+        duration *= samples_per_clk  # the duration here is expressed in samples
+        padding = samples_per_clk
```

Comment:
Maybe avoid the alias, and just use a single variable name (even `padding`, but from the beginning). At least it's easier to understand that those two are the same variable.